README.md

```diff
@@ -47,6 +35,83 @@ we aim to minimize our dependencies, but still rely on ones such as
 - build-essential (build tools for c/c++)
 
 as well as some others.
+
+# Docker installation
+for ease of use, we recommend you to use this method.
+
+all the dependencies are all retrieved by and contained within docker containers. all you need to install on your system is docker and docker-compose, and ensure that your user is a member of the docker group. if your package manager doesn't do that for you, you may need to log out and back in.
+## installing bancho.py's requirements
+```sh
+# install docker and docker-compose
+sudo apt install -y docker \
+                    docker-compose
+```
+## download the osu! server codebase onto your machine
+```sh
+# clone bancho.py's repository
+git clone https://github.com/osuAkatsuki/bancho.py
+
+# enter bancho.py's new directory
+cd bancho.py
+```
+## configuring bancho.py
+all configuration for the osu! server (bancho.py) itself can be done from the
+`.env` file. we provide an example `.env.example` file which you can use as a base.
+```sh
+# create a configuration file from the sample provided
+cp .env.example .env
+
+# you'll want to configure *at least* the three marked (XXX) variables,
+# as well as set the OSU_API_KEY if you need any info from osu!'s v1 api
+# (e.g. beatmaps).
+
+# open the configuration file for editing
+nano .env
+```
+## configuring a reverse proxy (we'll use nginx)
+```sh
+# copy the example nginx configuration file
+cp ext/nginx.conf.example ext/nginx.conf
+
+# now, you can edit the config file.
+# the spots you'll need to change are marked.
+nano ext/nginx.conf
+```
+
+## congratulations! you just set up an osu! private server
+
+if everything went well, you should be able to start your server up:
+
+```sh
+# start the server. note: you may have to start the services
+# one-by-one until some future update does it automatically:
```

Comment:
i should be able to set this up. is it an issue of race conditions?

Reply:
i think I've managed to fix it, check it when you are free